addons/payment_demo/tests/test_payment_transaction.py

```diff
@@ -100,3 +100,10 @@ def test_send_partial_capture_request_creates_capture_tx(self):
         self.assertEqual(
             child_tx.amount, 10, msg="Child transaction should have the requested amount."
         )
+
+    def test_compare_notification_data_without_provider_code_throws(self):
+        """ Test that comparing notification data without a provider code throws an error. """
+        self.provider.code = 'none'
+        tx = self._create_transaction('direct')
+        with self.assertRaises(NotImplementedError):
+            tx._compare_notification_data(self.notification_data)
```

Comment:
I'd drop this; it has no value as a test for Demo.

A better test could be to test that the base `_compare_notification_data` does raise, so it will require every provider module to override it.